src/store/useDarkModeStore.js

```diff
@@ -0,0 +1,30 @@
+import { create } from "zustand";
+
+const useDarkModeStore = create((set) => ({
+  isDark: localStorage.getItem("theme") === "dark",
+  toggleDark: () =>
+    set((state) => {
+      const newTheme = !state.isDark;
+      if (newTheme) {
+        document.documentElement.classList.add("dark");
+        localStorage.setItem("theme", "dark");
+      } else {
+        document.documentElement.classList.remove("dark");
+        localStorage.setItem("theme", "light");
+      }
+      return { isDark: newTheme };
+    }),
+  setDark: (value) =>
+    set(() => {
+      if (value) {
+        document.documentElement.classList.add("dark");
+        localStorage.setItem("theme", "dark");
+      } else {
+        document.documentElement.classList.remove("dark");
+        localStorage.setItem("theme", "light");
+      }
+      return { isDark: value };
+    }),
```

Comment:
또 `toggleDark`와 `setDark` 함수 내부에는 동일한 로직이 중복되어있는 거 같아요. 이렇게 리팩토링하면 좀 더 깔끔한 코드가 될 거 같습니다!
```suggestion
const applyTheme = (isDark) => {
  if (isDark) {
    document.documentElement.classList.add("dark");
    localStorage.setItem("theme", "dark");
  } else {
    document.documentElement.classList.remove("dark");
    localStorage.setItem("theme", "light");
  }
};

const useDarkModeStore = create((set) => ({
  isDark: localStorage.getItem("theme") === "dark",
  toggleDark: () =>
    set((state) => {
      const newTheme = !state.isDark;
      applyTheme(newTheme);
      return { isDark: newTheme };
    }),
  setDark: (value) =>
    set(() => {
      applyTheme(value);
      return { isDark: value };
    }),
}));
```

Reply:
그렇네요 다시 보니 코드가 중복되는군요!